ouroboros-consensus/docs/AddingAnEra.md

```diff
@@ -0,0 +1,126 @@
+# Adding a new Shelley-based era to consensus
+
+This document walks through the steps required to add a new Shelley-based era to
+consensus and to add it as an extra era to the Cardano blockchain.
+
+Prior art upon which this is based:
+* [#2666](https://github.com/input-output-hk/ouroboros-network/pull/2666)
+* [#2679](https://github.com/input-output-hk/ouroboros-network/pull/2679)
+
+The steps are fairly straightforward. We have put things in place (in consensus
+and in the ledger) when adding the Allegra and Mary eras that should now make it
+much easier to add a new additional era.
+
+There are two main driving changes: adding the new era to
+`Ouroboros.Consensus.Shelley.Eras` and including that era in
+`Ouroboros.Consensus.Cardano.Block`. The compiler should point the rest of the
+way. It's mostly a matter of following the existing patterns by imitating the
+previous Shelley-based case. Be sure to run both the Shelley
```

Comment:
> The compiler should point the rest of the
way. It's mostly a matter of following the existing patterns by imitating the
previous Shelley-based case.

🙌